build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+#       "driver=${DRIVER}" are added automatically.
+#   XFAIL: When set for x86_64|arm64-v8a|rv64, the test is expected to
+#       fail for the particular platforms/architectures for various reasons.
+#       e.g., benchmark_suite module is not available, upstream llvm backend
+#       breakage, etc.
+#   TIMEOUT: (optional) Test timeout.
+#
+# Example:
+#
+# iree_run_module_test(
+#   NAME
+#     person_detect_int8_correctness_test
+#   BENCHMARK_MODULE_SRC
+#     "TFLite/PersonDetect-int8"
+#   EXPECTED_OUTPUT
+#     "1x2xi8=[72 -72]"
+#   XFAIL
+#     "arm64-v8a"
+# )
+function(iree_run_module_test)
+  if(NOT IREE_BUILD_TESTS)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE
+    ""
+    "NAME;BENCHMARK_MODULE_SRC;MODULE_SRC;DRIVER;EXPECTED_OUTPUT;TIMEOUT"
+    "RUNNER_ARGS;LABELS;XFAIL"
+    ${ARGN}
+  )
+
+  if(CMAKE_CROSSCOMPILING AND "hostonly" IN_LIST _RULE_LABELS)
+    return()
+  endif()
+
+  if(_RULE_BENCHMARK_MODULE_SRC AND _RULE_MODULE_SRC)
+    message(SEND_ERROR "Can only specify either MODULE_SRC or BENCHMARK_MODULE_SRC")
+  endif()
+
+  if(NOT _RULE_BENCHMARK_MODULE_SRC AND NOT _RULE_MODULE_SRC)
+    message(SEND_ERROR "Need to specify either MODULE_SRC or BENCHMARK_MODULE_SRC")
+  endif()
+
+  # Benchmark suite needs to be stored at the location of
+  # `IREE_BENCHMARK_SUITE_DIR` or the test target is bypassed.
+  if(_RULE_BENCHMARK_MODULE_SRC AND NOT DEFINED IREE_BENCHMARK_SUITE_DIR)
+    return()
+  endif()
+
+  set(_RUNNER_TARGET "iree-run-module")
+  iree_package_name(_PACKAGE_NAME)
+  set(_NAME "${_PACKAGE_NAME}_${_RULE_NAME}")
+
+  if(_RULE_XFAIL)
+    platform_xfail(_XFAIL ${_RULE_XFAIL})
+  endif()
+
+  if(_RULE_MODULE_SRC)
+    get_filename_component(_SRC "${_RULE_MODULE_SRC}" REALPATH)
+  elseif(_RULE_BENCHMARK_MODULE_SRC)
+    set(_MODULE_FLAG_DIR "${IREE_BENCHMARK_SUITE_DIR}/${_RULE_BENCHMARK_MODULE_SRC}/")
+    # Find the platform specific module flag file with matching path name.
+    # TODO(#10391): Update this logic with the new benchmark framework.
+    if(CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+       CMAKE_SYSTEM_NAME STREQUAL "Linux")
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*RV64*__full-inference,default-flags/flagfile")
+    elseif(ANDROID)
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*ARM64-v8A*__big-core,full-inference,default-flags/flagfile")
+    else()  # X86_64
+      set(_FLAGFILE_HINT_PATH "${_MODULE_FLAG_DIR}/iree-llvm-cpu*x86_64*__full-inference,default-flags/flagfile")
+    endif()
+    file(GLOB _FLAGFILE_PATH
+        LIST_DIRECTORIES FALSE
+        "${_FLAGFILE_HINT_PATH}"
+      )
+    if(_FLAGFILE_PATH)
+      get_filename_component(_FLAG_FILE_DIR "${_FLAGFILE_PATH}" DIRECTORY)
+      file(STRINGS "${_FLAGFILE_PATH}" _FLAGS ENCODING UTF-8)
+      # Parse the flagfile to find the vmfb location, entry_function,
+      # and function_input to append to the runner arguments.
```

Comment:
ewwwwww. Not sure we have a better option here. I assume the TODO above includes getting rid of this?

Reply:
yes.